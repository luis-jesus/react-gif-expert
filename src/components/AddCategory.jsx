import { useState } from "react";
//import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setinputValue] = useState('');

    const onInputchange = ({target}) => {
        setinputValue( target.value );
    };

    const onSubmit = ( event ) => {
        event.preventDefault();
        if(inputValue.trim().length <= 1) return;
        console.log(inputValue);
        //setCategories(categories => [ inputValue, ...categories ]);
        setinputValue('');
        onNewCategory( inputValue.trim())
    };

//<form onSubmit={(event) => onSubmit(event)}> poner submit con callback, solo aplica con un parametro
  return (
    <form onSubmit={onSubmit}>
        <input
          type="text"
          placeholder="Buscar gif"
          value={inputValue}
          onChange={onInputchange}
        />
    </form>
  );
}
