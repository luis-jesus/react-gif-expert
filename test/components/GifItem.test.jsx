import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GitItem/>', () => {
    const title = 'Spider-Man';
    const url = 'https://spider-man.com/';

    test('Debe de hacer match con el snapshot', () => {
        const { container } = render( <GifItem title={title} url={url} /> );
        expect( container ).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {

        render( <GifItem title={title} url={url} /> );

        //screen.debug();
        //expect( screen.getByRole('img').src).toBe( url );

        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );
        
    });

    test('Debe de mostrar el titulo del componente', () => {
        render( <GifItem title={title} url={url} /> );
        expect(screen.getByText( title )).toBeTruthy();        
    })
});