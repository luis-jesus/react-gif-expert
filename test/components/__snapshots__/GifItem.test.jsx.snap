// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GitItem/> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Spider-Man"
      src="https://spider-man.com/"
    />
    <p>
      Spider-Man
    </p>
  </div>
</div>
`;
